src/test/deploy/functions/services/auth.spec.ts

```diff
@@ -0,0 +1,495 @@
+        platform: "gcfv1",
+        id: "id2",
+        entryPoint: "func2",
+        blockingTrigger: {
+          eventType: BEFORE_SIGN_IN_EVENT,
+          options: {
+            accessToken: true,
+          },
+        },
+      };
+      const want: backend.Backend = {
+        ...backend.of(ep1, ep2),
+      };
+
+      expect(() => authBlockingService.validateTrigger(ep1, want)).to.not.throw();
+    });
+  });
+
+  describe("registerBlockingTrigger", () => {
+    it("should handle an empty config", async () => {
+      const blockingConfig = {};
+      const newBlockingConfig: identityPlatform.BlockingFunctionsConfig = {
+        triggers: {
+          beforeCreate: {
+            functionUri: "somethingnew.url",
+          },
+        },
+        forwardInboundCredentials: {
+          accessToken: false,
+          idToken: true,
+          refreshToken: false,
+        },
+      };
+      const ep: backend.Endpoint = {
+        ...BASE_EP,
+        platform: "gcfv1",
+        uri: "somethingnew.url",
+        blockingTrigger: {
+          eventType: BEFORE_CREATE_EVENT,
+          options: {
+            accessToken: false,
+            idToken: true,
+            refreshToken: false,
+          },
+        },
+      };
+      getConfig.resolves(blockingConfig);
+      setConfig.resolves(newBlockingConfig);
+
+      await authBlockingService.registerTrigger(ep);
+
+      expect(blockingConfig).to.deep.equal(newBlockingConfig);
+      expect(setConfig).to.have.been.calledWith("project", newBlockingConfig);
+    });
+
+    it("should register on a new beforeCreate endpoint", async () => {
+      const blockingConfig: identityPlatform.BlockingFunctionsConfig = {
+        triggers: {
+          beforeCreate: {
+            functionUri: "beforecreate.url",
+          },
+          beforeSignIn: {
+            functionUri: "beforesignin.url",
+          },
+        },
+        forwardInboundCredentials: {
+          accessToken: true,
+        },
+      };
+      const newBlockingConfig: identityPlatform.BlockingFunctionsConfig = {
+        triggers: {
+          beforeCreate: {
+            functionUri: "somethingnew.url",
+          },
+          beforeSignIn: {
+            functionUri: "beforesignin.url",
+          },
+        },
+        forwardInboundCredentials: {
+          accessToken: false,
+          idToken: true,
+          refreshToken: false,
+        },
+      };
+      const ep: backend.Endpoint = {
+        ...BASE_EP,
+        platform: "gcfv1",
+        uri: "somethingnew.url",
+        blockingTrigger: {
+          eventType: BEFORE_CREATE_EVENT,
+          options: {
+            accessToken: false,
+            idToken: true,
+            refreshToken: false,
+          },
+        },
+      };
+      getConfig.resolves(blockingConfig);
+      setConfig.resolves(newBlockingConfig);
+
+      await authBlockingService.registerTrigger(ep);
+
+      expect(blockingConfig).to.deep.equal(newBlockingConfig);
```

Comment:
huh I'm having a brain freeze, but how does this pass? `blockingConfig` and `newBlockingConfig` clearly looks different to me.

Reply:
Agreed. I think this is happening because we're mutating return values of blockingConfig? Which seems wrong here... Maybe `setConfig` should resolve w/o any return value (unless we depend on it) and we shoudl be verifying that `setConfig` was called with `newBlockingConfig`.

Reply:
fixed in #4443, we no longer check for a modification to the object